client/src/sim.rs

```diff
@@ -305,6 +305,14 @@ impl Sim {
                 tracing::error!("Voxel data received from server for ungenerated chunk.")
             }
         }
+        for (chunk_id, voxel_data) in msg.modified_chunks {
+            let Some(voxel_data) = VoxelData::from_serializable(&voxel_data, self.cfg.chunk_size)
+            else {
+                tracing::error!("Voxel data received from server is invalid");
```

Comment:
s/invalid/of incorrect dimension/

Never be vague in an error message if you can possibly avoid it.